### DISCUSSION TITLE
How do I move trivy-db to my local image registry (Harbor)? OCI

### DISCUSSION THREAD

**Discussion by GitLabOps** [aquasecurity/trivy#4113] [Q&A]
Hello, everyone!
Has anyone tried using utilities like crane, skopeo, oras to move Trivy-db from ghcr.io/aquasecurity/trivy-db:2 to harbor.prod.com?

Using pull and then push methods for these commands looks too redundant in pipelines. 

I would like to use the following configuration for these commands:

- crane cp
- skopeo cp
- oras cp

**Take oras as an example:**
oras cp ghcr.io/aquasecurity/trivy-db:2 harbor.prod.com/public/aquasec/trivy/trivy-db:2
**Output:**
_Exists  85e7f991e22f db.tar.gz
Error: PUT "https://harbor.prod.com/v2/public/aquasec/trivy/trivy-db/blobs/uploads/8b6decdd-6ff9-466c-b5cd-3a2d8557cd14?_state=x6aAuvNrITUGRQuLQUj15w__tBUuzFTB9VPWFAgxsmd7Ik5hbWUiOiJwdWJsaWMvYXF1YXNlYy90cml2eS90cml2eS1kYiIsIlVVSUQiOiI4YjZkZWNkZC02ZmY5LTQ2NmMtYjVjZC0zYTJkODU1N2NkMTQiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMjMtMDQtMjRUMTA6MDk6NDcuMjE1NTYxNTUyWiJ9&digest=sha256%3Ae3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855": response status code 500: unknown: unknown error: map[RequestFailure:map[]]_

**Now I'm trying it this way:**
oras cp ghcr.io/aquasecurity/trivy-db:2 harbor.prod.com/public/aquasec/trivy/trivy-db:2 --to-oci-layout
**Output:**
_Copying 85e7f991e22f db.tar.gz
Copied  85e7f991e22f db.tar.gz
Copied [registry] ghcr.io/aquasecurity/trivy-db:2 => [oci-layout] harbor.prod.com/public/aquasec/trivy/trivy-db:2
Digest: sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e_

And this design works fine, but if I want to do it at the direction of the digests - problems appear:
crane digest ghcr.io/aquasecurity/trivy-db:2
**Output:**
sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e

oras cp ghcr.io/aquasecurity/trivy-db@sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e harbor.prod.com/public/aquasec/trivy/trivy-db@sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e --to-oci-layout
**Output:**
Copied [registry] ghcr.io/aquasecurity/trivy-db@sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e => [oci-layout] harbor.prod.com/public/aquasec/trivy/trivy-db:2
Digest: sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e

oras pull harbor.prod.com/public/aquasec/trivy/trivy-db@sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e
**Output:**
Error: failed to resolve sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e: harbor.prod.com/public/aquasec/trivy/trivy-db@sha256:c5936028d6cc5db68fe4a324aa3b6b2419a4947b3c67b7ca5943aab25f51276e: not found

If anyone has already encountered a solution to such a problem, please share. Let's add it to the documentation, if possible and with community approval.

I don't understand how I can make a normal oras cp so that I can provide a transparent supply chain of the trivi base to my private registry.

If anyone has already encountered a solution to such a problem, please share. Let's add it to the documentation, if possible and with community approval.

**Comment by xiaogangwen** [comment]
It might be backend object storage or driver issue. I got similar error message when using `oras push` for an OCI image to a registry backed by Minio.
